### ISSUE TITLE
Issue on Android R

### ISSUE THREAD

**Issue by DrKetan** [iBotPeaches/Apktool#2421]
### Information
1. **Apktool Version (``)** - 2.4.1
2. **Operating System (Mac, Linux, Windows)** - Window 7 32bit
3. **APK From? (Playstore, ROM, Other)** - ROM (Android R Beta Devce Samsung N20 ultra)

### Stacktrace/Logcat
Targeting R+ (version 30 and above) requires the resources.arsc of installed APKs to be stored uncompressed and aligned on a 4-byte boundary
```
Include stacktrace here
```
No error building apk but App doesn't get registered to system , found some documentation
https://developer.android.com/about/versions/11/behavior-changes-11

### Steps to Reproduce
1. `apktool `

### Frameworks
If this APK is from an OEM ROM (Samsung, HTC, LG). Please attach framework files
(`.apks` that live in `/system/framework` or `/system/priv-app`)

### APK
If this APK can be freely shared, please upload/attach a link to it.

[Files.zip](https://github.com/iBotPeaches/Apktool/files/5457265/Files.zip)

### Questions to ask before submission
1. Have you tried `apktool d`, `apktool b` without changing anything? - Yes
2. If you are trying to install a modified apk, did you resign it? Yes
3. Are you using the latest apktool version? Yes

**Comment by BurgerZ** [comment]
just do `zipalign -f -p 4 input.apk output.apk` after you compile the apk.

**Comment by iBotPeaches** [comment]
We have an existing ticket for R and zipaligning to align bytes is preferred.

**Comment by ElDavoo** [comment]
Using zipalign does not help, since:
resources.arsc of installed APKs to be stored uncompressed AND aligned on a 4-byte boundary
But apktool builds a compressed apk, so it fails anyway.

**Comment by prasannakumark** [comment]
Hi @ElDavoo 
Could you please more information on what you've edited? and how you fixed it?
I just tried removing - resources.arsc from apktool.yml but unfortunately, it's not working.

Thanks in advance!!

**Comment by ElDavoo** [comment]
Hi,
[The apk i've edited](https://drive.google.com/file/d/1lubC1WgJIMHSBto9DKXwkbSYLTD_r0eK/view?usp=sharing)
Since it was not installing the error above, i wanted to use apktool to extract and recompress the apk, leaving resources.arsc uncompressed. However, since resources.arsc was compressed in the original apk, it was not automatically added to the doNotCompress option of apktool.yml . For this reason, when I recompressed the apk unzip and zipalign showed that the file was compressed. Adding "arsc" to the doNotCompress option of apktool.yml solved the problem because the apk was rebuild and resources.arsc was not compressed, allowing me to install the application (which crashes anyway but it's a problem of the app).For this reason, I propose that when the cmdline "-api 30" is written, apktool systematically adds resources.arsc to the doNotCompress option of apktool.yml. This way, the error will never appear.
Tell me if there is anything unclear.;
In short: apktool only behaves incorrectly only with apks that are already wrong.

**Comment by steadfasterX** [comment]
for those (like me) who have not much exp with app dev (yet):

I stumbled over this error as well and I **did** the zipalign part and for sure my resources.arsc was **not** compressed (zipalign -v shows that!). The reason for me was that I still used `jarsigner` from an older guide I had.. 
so the fix was simply using `apksigner` instead and it worked immediately ( [source](https://stackoverflow.com/a/68870448/14611530) )
